acceptance/suites/tests/certificate_authority/intermediate_ca.rb

```diff
@@ -88,38 +98,46 @@ def fixture(filename)
 
-  # Remove the old CA infrastructure from an agent
-  on test_agent, "rm -rf #{test_agent.puppet['ssldir']}"
+  step 'Remove the old CA infrastructure from an agent' do
+    ssldir = puppet_config(test_agent, 'ssldir', section: 'agent')
+    on test_agent, "rm -rf #{ssldir}"
+  end
 
-  ssl_options = "--cert #{master.puppet['hostcert']} " +
-                "--cacert #{master.puppet['localcacert']} " +
-                "--key #{master.puppet['hostprivkey']}"
+  # We use the master's PKI with curl since it won't change again until after
+  # the test, though all curl calls below need to happen on the master.
+  cert = puppet_config(master, 'hostcert', section: 'master')
+  cacert = puppet_config(master, 'localcacert', section: 'master')
+  key = puppet_config(master, 'hostprivkey', section: 'master')
+  ssl_options = "--cert #{cert} --cacert #{cacert} --key #{key}"
+
+  step 'Download the intermediate PKI and submit a test CSR' do
+    on test_agent,
+      puppet("agent -t --certname fake_agent_name --server #{master}"),
+      :acceptable_exit_codes => [1]
+  end
 
-  # Create a dummy agent
-  on test_agent,
-    puppet("agent -t --certname fake_agent_name --server #{master}"),
-    :acceptable_exit_codes => [1]
+  step 'Sign the cert and verify the agent runs correctly with its new PKI' do
+    on master, "curl -XPUT #{ssl_options} " +
+               %q<-H 'Content-Type: application/json' -d '{"desired_state":"signed"}' > +
+               "--url https://#{master}:8140/puppet-ca/v1/certificate_status/fake_agent_name?environment=production"
 
-  # Sign the cert
-  on master, "curl -XPUT #{ssl_options} " +
-             %q<-H 'Content-Type: application/json' -d '{"desired_state":"signed"}' > +
-             "--url https://#{master}:8140/puppet-ca/v1/certificate_status/fake_agent_name?environment=production"
-  on test_agent,
-     puppet("agent -t --certname fake_agent_name --server #{master} --noop")
+    on test_agent,
+       puppet("agent -t --certname fake_agent_name --server #{master} --noop")
+  end
 
 
-  # Revoke the cert -- We need to be using the master for this!
-  on master, "curl -XPUT #{ssl_options} " +
-             %q<-H 'Content-Type: application/json' -d '{"desired_state":"revoked"}' > +
-             "--url https://#{master}:8140/puppet-ca/v1/certificate_status/fake_agent_name?environment=production"
+  step 'Verify revocation works as expected' do
+    on master, "curl -XPUT #{ssl_options} " +
+               %q<-H 'Content-Type: application/json' -d '{"desired_state":"revoked"}' > +
+               "--url https://#{master}:8140/puppet-ca/v1/certificate_status/fake_agent_name?environment=production"
 
-  on master, "curl -XDELETE -H 'Content-Type: application/json' #{ssl_options} " +
-             "--url https://#{master}:8140/puppet-ca/v1/certificate_status/fake_agent_name?environment=production"
+    on master, "curl -XDELETE -H 'Content-Type: application/json' #{ssl_options} " +
```

Comment:
what's the purpose of the delete here?

Reply:
Revoking a cert doesn't remove it from the CA's signed certs directory, this will delete the revoked cert as well.